prow/tests/config_test.go

```diff
@@ -0,0 +1,64 @@
+/*
+Copyright 2021 Google LLC
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tests
+
+import (
+	"strings"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/util/sets"
+)
+
+func TestInrepoConfigSettings(t *testing.T) {
+	irc := c.InRepoConfig
+
+	allowedOrgRepo := sets.NewString()
+	for or, b := range irc.Enabled {
+		if b != nil && *b {
+			allowedOrgRepo.Insert(or)
+		}
+	}
+
+	for or, clusters := range irc.AllowedClusters {
+		var private bool
+		for _, cluster := range clusters {
+			if strings.Contains(cluster, "private") {
+				private = true
+				break
+			}
+		}
+		if !private {
+			continue
+		}
+		var org, repo string
+		// Normalizing with gerrit org name like `https://org.something.com`
+		for _, gerritPrefix := range []string{"https://", "http://"} {
+			if strings.HasPrefix(or, gerritPrefix) {
+				org = gerritPrefix
+				or = strings.Replace(or, gerritPrefix, "", 1)
+			}
+		}
+		parts := strings.SplitN(or, "/", 2)
+		if len(parts) > 1 {
+			repo += parts[1]
+		}
+		org += parts[0]
+		if !allowedOrgRepo.Has(org) && !allowedOrgRepo.Has(org+"/"+repo) {
+			t.Fatalf("Org/repo not allowed to use private cluster. Want: %q enable inrepoconfig, got: %v", or, allowedOrgRepo.List())
```

Comment:
If I understand correctly this validates that in repo config is enabled for any org or repo that allows a private build cluster for in repo config jobs. I don't think this is something we need to check. It doesn't make sense for an org or repo to configure allowed build clusters for inrepo config if they don't even have inrepo config enabled so that should be checked by `checkconfig` if it isn't already.

I think what we actually need to be validating is that no jobs in this repo run in build clusters that include `private` with the exception of the `checkconfig` jobs.

Reply:
If we want to keep this validation it should be fixed to no longer be specific to private repos and we should upstream it.

Reply:
I'll just drop this